### ISSUE TITLE
Firestore doesn't work

### ISSUE THREAD

**Issue by JOELwindows7** [WolfgangSenff/GodotFirebase#33]
I have problem when doing with Firebase Database. I have been figuring out what was going on with it, by analyzing this addon, for why doesn't it work.

The Auth here works, but The Firestore no longer works. I even tried the demo if I made wrong, but infact **The demo also no longer works for updloading firebase**

I have tried different method to do with Firebase, by trying the simpler script found in GDQuest tutorial [here](https://youtu.be/-vDNk7BzOGc ). and its finished file right [here](https://github.com/GDQuest/godot-demos/blob/master/2019/05-21-firebase-firestore/end/static/Firebase.gd ). Here I created another script to become the Singleton under different node name, let's say `FirebaseQuest`.

**and GDQuest version works right away (both Auth and Firestore), no problem**

here's example of `push` from GDQuest
```
func save_document(path: String, fields: Dictionary, http: HTTPRequest) -> void:
	var document := { "fields": fields }
	var body := to_json(document)
	var url := FIRESTORE_URL + path
	http.request(url, _get_request_headers(), false, HTTPClient.METHOD_POST, body)
```

and yours from `FirebaseReference.gd`
```
func push(data):
	var to_push = JSON.print(data)
	if pusher.get_http_client_status() == HTTPClient.STATUS_DISCONNECTED:
		pusher.request(_get_list_url() + db_path + _get_remaining_path(), PoolStringArray(), true, HTTPClient.METHOD_POST, to_push)
	else:
		push_queue.append(data)
```

Let me now comment what best I could understand. correct me if wrong, it's quite complicated for me to learn back this addon, I know.:
- GDquest version do concern about request headers in the custom header every HTTP request. Maybe now it is required to have custom header when uploading HTTP, since yours lack the custom header, I think Google rejects it from now on. 
- Note that I had Firestore security rule designed that only allow read and write if user logged in. And again, attempting writing to and reading from there, only GDquest version that works
- If the HTTP has finished doing its business (`HTTPClient.STATUS_DISCONNECTED`), well that mean it should be able to write so, since the http is not connected right now.
- GDQuest version place every data to be pushed or patched with into a dictionary of `{"fields":fields}` in which the `"fields"` contains all the data you'd like to push or update with. I think the upload structure might have changed, so it's no longer compatible with older technique.
- The structure of the data then should looks like
```
var tobePush = {
		"fields":{
			"Auza":{"stringValue":"aaaaaaaaaaaw"}
		},
	}
```
- notice the `stringValue` there. the datatype has to be defined first. Haven't seen again if your demo had one.
- So basically, you just do
```
var data = { "aaaa" : {"stringValue" : "sample" } }
FirebaseQuest.push(data)
```
and that's it. it works (assuming you abide by your Firestore rule e.g. logged in).

what's going on?

**Comment by WolfgangSenff** [comment]
You're actually looking in the wrong place. Firestore is not implemented via references. References are used only for the realtime database from Firebase. Look at FirebaseFirestore.gd, FirebaseFirestoreDocument.gd, and FirebaseFirestoreCollection.gd and you might get a better understanding of how to use it.

Also, GDQuest's version does not properly handle streamed updates, from what I remember. In the future, it would make sense not to look at their codebase and expect mine to behave the same way; I'm a professional Firebase developer, they are not.

**Comment by JOELwindows7** [comment]
I have issued #37 for continuation here as I mistakenly thought it didn't work while actually it works, **but it uses Realtime Database instead**
